framework/encode/vulkan_capture_manager.cpp

```diff
@@ -2822,5 +2823,34 @@ void VulkanCaptureManager::PreProcess_vkBindImageMemory2(VkDevice
     }
 }
 
+void VulkanCaptureManager::PostProcess_vkCreateShaderModule(VkResult                        result,
+                                                            VkDevice                        device,
+                                                            const VkShaderModuleCreateInfo* pCreateInfo,
+                                                            const VkAllocationCallbacks*    pAllocator,
+                                                            VkShaderModule*                 pShaderModule)
+{
+    GFXRECON_UNREFERENCED_PARAMETER(device);
+    GFXRECON_UNREFERENCED_PARAMETER(pAllocator);
+    GFXRECON_UNREFERENCED_PARAMETER(pShaderModule);
+
+    if (result == VK_SUCCESS)
+    {
+        // spirv-parsing for buffer-references
+        gfxrecon::util::SpirVParsingUtil spirv_util;
+
+        if (spirv_util.ParseBufferReferences(pCreateInfo->pCode, pCreateInfo->codeSize))
+        {
+            auto buffer_reference_infos = spirv_util.GetBufferReferenceInfos();
+
+            if (!buffer_reference_infos.empty())
+            {
+                GFXRECON_LOG_WARNING_ONCE(
+                    "Shader is using the 'SPV_KHR_physical_storage_buffer' feature. "
```

Comment:
It might be good to also add the warning to replay. From a user's perspective, replay may be where they see something go wrong so having the warning logged in the replay's output could be helpful.

Reply:
Added parsing + warning to vulkan_replay_consumer, issuing this warning if a replay hits a buffer-ref:

_[gfxrecon] WARNING - Shader is using the 'SPV_KHR_physical_storage_buffer' feature. Resource tracking for buffers accessed via references is currently unsupported, so replay may fail._